tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -0,0 +1,833 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Threading;
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Bson.TestHelpers.JsonDrivenTests;
+using MongoDB.Bson.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Configuration;
+using MongoDB.Driver.Core.Events;
+using MongoDB.Driver.Core.Misc;
+using MongoDB.Driver.Core.Operations;
+using MongoDB.Driver.Core.TestHelpers.XunitExtensions;
+using MongoDB.Driver.Encryption;
+using MongoDB.Driver.TestHelpers;
+using MongoDB.Libmongocrypt;
+using Xunit;
+
+namespace MongoDB.Driver.Tests.Specifications.client_side_encryption.prose_tests
+{
+    public class ClientEncryptionProseTests
+    {
+        #region static
+        private static readonly CollectionNamespace __collCollectionNamespace = CollectionNamespace.FromFullName("db.coll");
+        private static readonly CollectionNamespace __keyVaultCollectionNamespace = CollectionNamespace.FromFullName("admin.datakeys");
+        #endregion
+
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+        private const string SchemaMap =
+            @"{
+                ""db.coll"": {
+                    ""bsonType"": ""object"",
+                    ""properties"": {
+                        ""encrypted_placeholder"": {
+                            ""encrypt"": {
+                                ""keyId"": ""/placeholder"",
+                                ""bsonType"": ""string"",
+                                ""algorithm"": ""AEAD_AES_256_CBC_HMAC_SHA_512-Random""
+                              }
+                          }
+                      }
+                  }
+            }";
+
+        private readonly ICluster _cluster;
+        private readonly ICoreSessionHandle _session;
+
+        public ClientEncryptionProseTests()
+        {
+            _cluster = CoreTestConfiguration.Cluster;
+            _session = CoreTestConfiguration.StartSession(_cluster);
```

Comment:
In theory a session should be `Dispose`(d) when you are done using it.

It may not matter if the number of abandoned sessions is not very large...

Reply:
done